public/main.js

```diff
@@ -134,12 +147,23 @@ function deleteTodo(todoId, callback) {
     deleteRequest.open("DELETE", "/api/todo/" + todoId);
     deleteRequest.onload = function() {
         if (this.status === 200) {
-            callback();
         } else {
-            error.textContent = "Failed to delete item. Server returned " + this.status + " - " + this.responseText;
+            renderMessageDialog("error", "Failed to delete item " +
+                todoId.toString() + ". Server returned " + this.status + " - " + this.responseText);
         }
+        callback(this.status);
     };
     deleteRequest.send();
 }
 
+function deleteCompleted() {
+    var promises = [];
+    todosLocal.forEach(function(todo) {
+        if (todo.isComplete) {
+            promises.push(new Promise(function(resolve, reject) {deleteTodo(todo.id, resolve);}));
+        }
+    });
+    Promise.all(promises).then(reloadTodoList);
```

Comment:
If one of the requests fails, then the resolve won't be called and neither will reload. If some of the items were successfully deleted it won't be shown to the user. I think this will do for now, but it will have to be fixed later.